File: frontend/src/api/announcement.ts
import { apiCall } from "./client";

export interface AnnouncementRow {
  name: string;
  title: string;
  body: string;
  pinned: 0 | 1;
  published_on: string;
  published_by: string;
}

export const announcementApi = {
  feed: () => apiCall<AnnouncementRow[]>("GET", "fatehhr.api.announcement.feed"),
};

File: fatehhr/api/announcement.py
import frappe


@frappe.whitelist()
def feed(limit: int = 50) -> list[dict]:
	return frappe.get_all(
		"HR Announcement",
		fields=["name", "title", "body", "pinned", "published_on", "published_by"],
		order_by="pinned desc, published_on desc",
		limit=int(limit),
	)
